google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/ConnectionWorker.java

```diff
@@ -381,6 +471,103 @@ private void registerOpenTelemetryMetrics() {
+    instrumentAckedRequestCount =
+        writeMeter
+            .counterBuilder("append_requests_acked")
+            .setDescription("Counts number of requests acked by the server")
+            .build();
+    instrumentAckedRequestSize =
+        writeMeter
+            .counterBuilder("append_request_bytes_acked")
+            .setDescription("Counts byte size of requests acked by the server")
+            .build();
+    instrumentAckedRequestRows =
+        writeMeter
+            .counterBuilder("append_rows_acked")
+            .setDescription("Counts number of request rows acked by the server")
+            .build();
+    instrumentErrorRequestCount =
+        writeMeter
+            .counterBuilder("append_requests_error")
+            .setDescription("Counts number of requests returned by the server with an error")
+            .build();
+    instrumentErrorRequestSize =
+        writeMeter
+            .counterBuilder("append_request_bytes_error")
+            .setDescription("Counts byte size of requests returned by the server with an error")
+            .build();
+    instrumentErrorRequestRows =
+        writeMeter
+            .counterBuilder("append_rows_error")
+            .setDescription(
+                "Counts number of rows in requests returned by the server with an error")
+            .build();
+    writeMeter
+        .gaugeBuilder("waiting_queue_length")
+        .ofLongs()
+        .setDescription(
+            "Reports length of waiting queue. This queue contains requests buffered in the client and not yet sent to the server.")
+        .buildWithCallback(
+            measurement -> {
+              int length = 0;
+              this.lock.lock();
+              try {
+                length = waitingRequestQueue.size();
+              } finally {
+                this.lock.unlock();
+              }
+              measurement.record(length, getTelemetryAttributes());
+            });
+    writeMeter
+        .gaugeBuilder("inflight_queue_length")
+        .ofLongs()
+        .setDescription(
+            "Reports length of inflight queue. This queue contains sent append requests waiting for response from the server.")
+        .buildWithCallback(
+            measurement -> {
+              int length = 0;
+              this.lock.lock();
+              try {
+                length = inflightRequestQueue.size();
+              } finally {
+                this.lock.unlock();
+              }
+              measurement.record(length, getTelemetryAttributes());
+            });
+    instrumentNetworkResponseLatency =
+        writeMeter
+            .histogramBuilder("network_response_latency")
+            .ofLongs()
+            .setDescription(
+                "Reports time taken in milliseconds for a response to arrive once a message has been sent over the network.")
+            .setExplicitBucketBoundariesAdvice(METRICS_LATENCY_BUCKETS)
+            .build();
+    instrumentConnectionEstablishCount =
```

Comment:
I believe this can be derived from network_response_latency, if you put connection_id as the metrics field.

Reply:
I have added writer_id as an attribute. I still have this metric, however, as it directly provides information about establishing a connection.